src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -0,0 +1,84 @@
+package nextstep.subway.line.application;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import nextstep.subway.line.application.dto.LineRequest;
+import nextstep.subway.line.application.dto.LineResponse;
+import nextstep.subway.line.domain.LineRepository;
+import nextstep.subway.line.domain.entity.Line;
+import nextstep.subway.line.domain.entity.LineSection;
+import nextstep.subway.line.domain.entity.LineSections;
+import nextstep.subway.station.application.dto.StationResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.domain.StationRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+@AllArgsConstructor
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+
+        Line line = lineRepository.save(new Line(lineRequest.getName(), lineRequest.getColor(), new LineSections()));
+        LineSection lineSection = new LineSection(line, stationRepository.findByIdOrThrow(lineRequest.getUpStationId()), stationRepository.findByIdOrThrow(lineRequest.getDownStationId()), lineRequest.getDistance());
+
+        line.getLineSections().addSection(lineSection);
```

Comment:
값을 꺼내서 처리하는 대신에, 메시지를 보내서 처리해보면 어떨까요?
```suggestion
        line.addSection(lineSection);
```

Reply:
값 객체에 직접 접근하는 대신 메시지를 보내도록 수정하겠습니다!